game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/pbf/ForumPosterEditorViewModel.java

```diff
@@ -48,6 +48,28 @@
     populateFromGameProperties(properties);
   }
 
+  void setForumPassword(final char[] password) {
+    final ClientSetting<char[]> passwordSetting =
+        forumSelection.equals(NodeBbForumPoster.TRIPLEA_FORUM_DISPLAY_NAME)
+            ? ClientSetting.tripleaForumPassword
+            : ClientSetting.aaForumPassword;
+
+    passwordSetting.setValueAndFlush(password);
+    forumPassword = password;
+    readyCallback.run();
+  }
+
+  void setForumUsername(final String username) {
+    final ClientSetting<char[]> usernameSetting =
+        forumSelection.equals(NodeBbForumPoster.TRIPLEA_FORUM_DISPLAY_NAME)
+            ? ClientSetting.tripleaForumUsername
+            : ClientSetting.aaForumUsername;
+
+    usernameSetting.setValueAndFlush(username.toCharArray());
+    forumUsername = username;
+    readyCallback.run();
+  }
+
   @SuppressWarnings("Guava")
```

Comment:
Just noticed this: It should be easily removable now that guava's `Predicates#not` can be replaced with [`Predicate#not`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/function/Predicate.html#not(java.util.function.Predicate)) from the standard library

Reply:
Ah, indeed, thanks. The auto-correct for the warning was giving a nasty ".negation()"
Updated in: 668a0dfd1